e2e-tests/src/main.rs

```diff
@@ -248,3 +260,64 @@ fn accounts(seeds: Option<Vec<String>>) -> Vec<sr25519::Pair> {
     });
     seeds.into_iter().map(keypair_from_string).collect()
 }
+
+fn get_first_two_accounts(accounts: &[sr25519::Pair]) -> (sr25519::Pair, sr25519::Pair) {
+    let first = accounts.get(0).expect("No accounts passed").to_owned();
+    let second = accounts
+        .get(1)
+        .expect("Pass at least two accounts")
+        .to_owned();
+    (first, second)
+}
+
+#[derive(Debug)]
+struct FeeInfo {
+    fee_without_weight: Balance,
+    unadjusted_weight: Balance,
+    adjusted_weight: Balance,
+}
+
+fn get_tx_fee_info(connection: &Connection, tx: &TransferTransaction) -> FeeInfo {
+    let block = connection.get_block::<Block>(None).unwrap().unwrap();
+    let block_hash = block.header.hash();
+
+    let unadjusted_weight = connection
+        .get_payment_info(&tx.hex_encode(), Some(block_hash))
+        .unwrap()
+        .unwrap()
+        .weight as Balance;
+
+    let fee = connection
+        .get_fee_details(&tx.hex_encode(), Some(block_hash))
+        .unwrap()
+        .unwrap();
+    let inclusion_fee = fee.inclusion_fee.unwrap();
+
+    FeeInfo {
+        fee_without_weight: inclusion_fee.base_fee + inclusion_fee.len_fee + fee.tip,
+        unadjusted_weight,
+        adjusted_weight: inclusion_fee.adjusted_weight_fee,
+    }
+}
+
+fn balance_of(account: &AccountId32, connection: &Connection) -> Balance {
+    connection.get_account_data(account).unwrap().unwrap().free
+}
+
+fn transfer(target: &AccountId32, value: u128, connection: &Connection) -> TransferTransaction {
+    let tx: UncheckedExtrinsicV4<_> = compose_extrinsic!(
+        connection,
+        "Balances",
+        "transfer",
+        GenericAddress::Id(target.clone()),
+        Compact(value)
+    );
+
+    let tx_hash = connection
+        .send_extrinsic(tx.hex_encode(), XtStatus::InBlock)
```

Comment:
why not XtStatus::Finalized ?

Reply:
good question; I left it as it has already been, but I will change it

Reply:
changed